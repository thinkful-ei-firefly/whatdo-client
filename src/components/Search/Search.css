
.SearchPage{
    height: 120vh;
    width:100%;
    margin: 0;
    padding: 0;
    font-family: sans-serif;
    display: flex;
    background: url("Desktop-1.png") no-repeat fixed center;
    background-repeat: no-repeat;
    background-size:fixed;
    animation: slide 10s  infinite;
}

@keyframes slide {
    0% {
        background: url("topLeft.jpg") left;
    }
    25%{
        background: url("topRight.jpg") right;
    }
    50%{
        background: url("downLeft.jpg") center;
    }
    100%{
        background: url("downRight.jpg") left;
    }
}

@media (min-width: 800px) {
    .SearchPage {
        height: 85vh;
    }
}

.Search{
    /* height: 100%; */
    /* width: 600px; */
    margin: auto;
    padding:20px;
    display: flex;
    /* justify-content: space-around; */
    align-items: center;
    flex-direction: column;
    font-family: sans-serif;
    background: rgba(255, 255, 255, 0.74);
    border-radius: 10px;
    
    /* background-repeat: no-repeat; */
    /* background-size: 1450px ;
    padding-top: 1vh;
    padding-bottom:auto;
    margin :auto; */
    
}

.Search h2{
    /* max-width: 80%; */
    
    padding: 10px 0px ;
    text-align: center;
    margin: 0 auto;
    border-radius: 7px;
    text-shadow: 0px 0px 6px rgba(255, 255, 255, 1);
-webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
-moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
transition: all 0.4s ease 0s;
background: rgba(250, 248, 241, 0.281);
animation: type 10s steps(22) ;
    overflow: hidden;
    white-space: nowrap;
    border-right: 4px solid black;
    border-left: 4px solid black;
    max-width:22ch;
}

.Search h3{
    /* max-width: 80%; */
    
    padding: 10px;
    text-align: center;
    margin: auto;
    margin-top:10px;
    border-radius: 7px;
    text-shadow: 0px 0px 6px rgba(255, 255, 255, 1);
-webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
-moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
transition: all 0.4s ease 0s;
background: rgba(250, 248, 241, 0.666);


}

@keyframes type {
    0% {
        width: 0px;;
    }

    50% {
        width: 22ch;
    }
    60% {
        width: 0ch;;
    }
    100% {
        width: 22ch;
    }
}


.Search h4{
    /* max-width: 80%; */
    
    padding: 10px;
    text-align: center;
    margin:0 auto;
    border-radius: 7px 7px 0   0;
    text-shadow: 0px 0px 6px rgba(255, 255, 255, 1);
-webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
-moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
transition: all 0.4s ease 0s;
background: rgba(250, 248, 241, 0.666);

}

.form {
    margin-bottom: 20px;
    width: 80%;
    max-width: 350px;
    /* position: relative; */
    /* height: 50px; */
    /* background: blue; */
    /* overflow:hidden; */
    align-items: center;
    border-radius: 11px;
    /* height: 80vh; */
    /* background: rgb(250, 248, 241); */
}
@media (min-width: 800px) {
    .Search {
        /* height: 85vh; */
    }
    

  }

.form-section {
    margin: 20px  auto;
    width: 100%;
    position: relative;
    height: 50px;
    border-radius:0 0 7px 7px;
    z-index: 1;
    /* text-align: center; */
    /* background: blue; */
    /* overflow:hidden; */


}

.date input{
    border: 3px solid #487eb0
}

.checkbox__section{
    padding :20px auto 0px auto;
    margin:0px;
    height: 120px;
    display: flex;
    align-items: center;
    flex-wrap: wrap;
    align-content: center;
    /* height: 100%; */
    align-content: stretch;
    background: rgba(250, 248, 241, 0.666);
}

.checkbox__section input +  label  {
    width:100px;
    flex-grow:1;

}

.checkbox__section input + label  {
    /* display: inline-block; */
    /* align-items: center; */
    margin:0.2em;
    cursor: pointer;
    padding:  0.2em;
    
}



.checkbox__section input {
    display: none;
    
}

.checkbox__section input +label:before{
    content: "\2714";
    border:0.1em solid #000;
    border-radius: 0.2em;
    display: inline-block;
    width: 1em;
    height: 1em;
    padding-left:0.2em;
    padding-bottom:0.3em;
    margin-right:0.2em;
    vertical-align:bottom ;
    color:transparent;
    transition: .3s;
}

.checkbox__section input  + label:active:before {
    transform:scale(0);

}

.checkbox__section input:checked  + label:before{
    background-color: rgb(97, 132, 248) ;
    border-color:rgb(97, 132, 248);
    color:#fff;
}

.checkbox__section input:disabled  + label:before{
    transform:scale(1);
    border-color:#aaa;

}

checkbox__section input:checked:disabled  + label:before{
    transform:scale(1);
    border-color:rgb(157, 199, 250);
    background-color:rgb(157, 199, 250); ;
}


.input {
    width: 100%;
    height: 100%;
    color: grey;
    padding-top: 10px;
    font-size: 20px;
    border: none;
    text-align: center;
    outline: none;
    border-radius: 50px;
     text-align: center;

}
.label-name {
    position: absolute;
    bottom:0px;
    left:0%;
    width: 100%;
    height: 100%;
    pointer-events: none;
    border-bottom: 2px solid lightcoral;
    text-align: center;
    border-radius: 50px;

}

.label-name::after {
    content: '';
    position: absolute;
    left:0px;
    bottom:-2px;
    height: 100%;
    width:100%;
    border-bottom: 4px solid #487eb0;
    transform: translateY(-3%);
    transition: transform 0.3s ease;
    border-radius: 50px;
 
}

.content-name {
    position: absolute;
    bottom:5px;
    left:0px;
    right:0px;
    transition:  all 0.3s ease;
    

}



.input:focus + .label-name .content-name,
.input:valid + .label-name .content-name {
    transform: translateY(-150%);
    font-size:14px;
    color:#487eb0;
    
}

.input:focus + .label-name::after,
.input:valid + .label-name::after {
    transform: translateX(0%);

}



.form-btn {
    width: 60%;
    text-align: center;
    margin-left:auto;
    margin-right:auto;
    align-items: center;
}


.searchForm__btn {
    margin: 20px;
    width: 100%;
    border: none;
    color: #fff !important;
    text-transform: uppercase;
    text-decoration: none;
    background: #487eb0;
    padding: 10px;
    border-radius: 50px;
    display: inline-block;
    border: none;
    transition: all 0.4s ease 0s;
    text-align: center;
    margin-left:auto;
    margin-right:auto;
    align-items: center;
    font-size: large;
    
}

.searchForm__btn:hover {
    text-shadow: 0px 0px 6px rgba(255, 255, 255, 1);
-webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
-moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
transition: all 0.4s ease 0s;
    }



