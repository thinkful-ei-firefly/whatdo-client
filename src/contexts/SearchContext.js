import React from 'react'

const SearchContext = React.createContext({
  city: 'Tucker',
  zipCode: '30084',
  weather: [],
  date: new Date(Date.now()),
  events: [],
  savedEvents: [],
  distance: 10,
  categories: '',
  music: false,
  attractions: false,
  bars: false,
  sports: false,
  performingArts: false,
  festivals: false,
  searchedTerms: {
    zipCode: '30084',
    distance: 10,
    date: new Date(Date.now()),
    music: false,
    attractions: false,
    bars: false,
    sports: false,
    performingArts: false,
    festivals: false,
  },
  apiSearch: () => {},
  getSavedEvents: () => {},
  saveEvent: () => {},
  removeEvent: () => {},
  setSearchTerm: () => {},
  setChecked: () => {},
  setZipCode: () => {},
  pageNum: 1,
  pageCount: 1,
  prevPage: () => {},
  nextPage: () => {},
  loading: false,
  toggleLoading: () => {},
  fetchError: false
})



export default SearchContext