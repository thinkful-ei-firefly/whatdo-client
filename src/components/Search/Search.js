import React from 'react';
import './Search.css';
import SearchContext from '../../contexts/SearchContext';
import CurrLocBtn from '../CurrLocBtn/CurrLocBtn';

export default class Search extends React.Component {
  static contextType = SearchContext;

  handleSubmit(ev) {
    ev.preventDefault();
    const zipCode = this.context.zipCode;
    const distance = this.context.distance;
    const date = new Date(`${this.context.date}T12:00:00`);

    let categories = [];

    for (let i = 1; i <= 6; i++) {
      if (this.context[`box${i}`].checked) {
        const category = `${this.context[`box${i}`]}`;
        categories.push(category);
      }
    }
    categories = categories.join();

    if (!categories)
      categories =
        'music,sports,outdoors_recreation,attractions,performing_arts,comedy,food,singles_social,festivals_parades,holiday';

    this.context.apiSearch(zipCode, distance, date, categories);
    //once SearchResults page is hooked up to context, modify apiSearch to return a promise
    //then add a .then chain to history.push the user to the results page
    this.props.handleSubmit();
  }

  render() {
    return (
      <div className="SearchPage">
        <div className="Search">
          <h2>Search event near you</h2>
          <h3>
            "Have fun, be yourself, enjoy life and stay positive" - Tatiana
            Maslany
          </h3>

          <form className="form" onSubmit={ev => this.handleSubmit(ev)}>
            {/*
              <div role="alert" className="alert">
                {error && <p>{error}</p>}
              </div>
            */}

            <div className="form-section">
              <input
                type="text"
                className="input"
                id="zipCode"
                name="zipCode"
                autoComplete="off"
                value={this.context.zipCode}
                onChange={this.context.setSearchTerm}
                required
              />
              <label className="label-name" htmlFor="zipCode">
                <span className="content-name">Enter your zipcode</span>
              </label>
              <CurrLocBtn />
            </div>

            <div className="form-section">
              <input
                type="text"
                className="input"
                id="distance"
                name="distance"
                autoComplete="off"
                value={this.context.distance}
                onChange={this.context.setSearchTerm}
                required
              />
              <label className="label-name" htmlFor="distance">
                <span className="content-name">Miles around you</span>
              </label>
            </div>

            <div className="form-section date">
              <input
                className="input"
                type="date"
                id="date"
                name="date"
                value={this.context.date}
                onChange={this.context.setSearchTerm}
              />
            </div>
            <h4>Filter by event type</h4>

            <div className="form-section checkbox__section">
              <input
                type="checkbox"
                id="box1"
                name="music"
                value="music"
                checked={this.context.music}
                onChange={this.context.setChecked}
              ></input>
              <label htmlFor="box1">Music</label>
              <input
                type="checkbox"
                id="box2"
                name="sports"
                value="sports,outdoors_recreation"
                checked={this.context.sports}
                onChange={this.context.setChecked}
              ></input>
              <label htmlFor="box2">Sports</label>
              <input
                type="checkbox"
                id="box3"
                name="attractions"
                value="attractions"
                checked={this.context.attractions}
                onChange={this.context.setChecked}
              ></input>
              <label htmlFor="box3">Attractions</label>
              <input
                type="checkbox"
                id="box4"
                name="performingArts"
                value="performing_arts,comedy"
                checked={this.context.performingArts}
                onChange={this.context.setChecked}
              ></input>
              <label htmlFor="box4">Performing Arts</label>
              <input
                type="checkbox"
                id="box5"
                name="bars"
                value="food,singles_social"
                checked={this.context.bars}
                onChange={this.context.setChecked}
              ></input>
              <label htmlFor="box5">Bars</label>
              <input
                type="checkbox"
                id="box6"
                name="festivals"
                value="festivals_parades,holiday"
                checked={this.context.festivals}
                onChange={this.context.setChecked}
              ></input>
              <label htmlFor="box6">Festivals</label>
            </div>

            <div className="form-btn">
              <button className="searchForm__btn" type="submit">
                Search
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
