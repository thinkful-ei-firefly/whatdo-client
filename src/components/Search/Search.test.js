import React from 'react'
import ReactDOM from 'react-dom'
import Search from './Search'
import renderer from 'react-test-renderer'
import SearchContext from '../../contexts/SearchContext'

describe('<Search />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<Search />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders as expected', () => {
    const value = {date:'12/12/12'}
    const tree = renderer
      .create(
        <SearchContext.Provider value={value}>
          <Search />
        </SearchContext.Provider>
      )
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
