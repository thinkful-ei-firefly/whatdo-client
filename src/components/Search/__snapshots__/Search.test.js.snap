// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders as expected 1`] = `
<div
  className="SearchPage"
>
  <div
    className="Search"
  >
    <h2>
      Search event near you
    </h2>
    <h3>
      "Have fun, be yourself ,enjoy life and stay positive" - Tatiana Maslany
    </h3>
    <form
      className="form"
      onSubmit={[Function]}
    >
      <div
        className="form-section"
      >
        <input
          autoComplete="off"
          className="input"
          id="zipCode"
          name="zipCode"
          required={true}
          type="text"
        />
        <label
          className="label-name"
          htmlFor="zipCode"
        >
          <span
            className="content-name"
          >
            Enter your zipcode
          </span>
        </label>
      </div>
      <div
        className="form-section"
      >
        <input
          autoComplete="off"
          className="input"
          id="distance"
          name="distance"
          required={true}
          type="text"
        />
        <label
          className="label-name"
          htmlFor="distance"
        >
          <span
            className="content-name"
          >
            Miles around you
          </span>
        </label>
      </div>
      <div
        className="form-section date"
      >
        <input
          className="input"
          defaultValue="2019-10-21"
          id="date"
          max="2019-12-31"
          min="2019-01-01"
          name="date"
          type="date"
        />
      </div>
      <h4>
        Filter by event type
      </h4>
      <div
        className="form-section checkbox__section"
      >
        <input
          id="box1"
          name="categories"
          type="checkbox"
          value="music"
        />
        <label
          htmlFor="box1"
        >
          Music
        </label>
        <input
          id="box2"
          name="categories"
          type="checkbox"
          value="sports,outdoors_recreation"
        />
        <label
          htmlFor="box2"
        >
          Sports
        </label>
        <input
          id="box3"
          name="categories"
          type="checkbox"
          value="attractions"
        />
        <label
          htmlFor="box3"
        >
          Attractions
        </label>
        <input
          id="box4"
          name="categories"
          type="checkbox"
          value="performing_arts,comedy"
        />
        <label
          htmlFor="box4"
        >
          Performing Arts
        </label>
        <input
          id="box5"
          name="categories"
          type="checkbox"
          value="food,singles_social"
        />
        <label
          htmlFor="box5"
        >
          Bars
        </label>
        <input
          id="box6"
          name="categories"
          type="checkbox"
          value="festivals_parades,holiday"
        />
        <label
          htmlFor="box6"
        >
          Festivals
        </label>
      </div>
      <div
        className="form-btn"
      >
        <button
          className="searchForm__btn"
          type="submit"
        >
          Search
        </button>
      </div>
    </form>
  </div>
</div>
`;
